keys/signing.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keys
+
+import (
+	"crypto/ed25519"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/coinbase/rosetta-sdk-go/asserter"
+	"github.com/coinbase/rosetta-sdk-go/types"
+	"github.com/ethereum/go-ethereum/crypto/secp256k1"
+)
+
+// signECDSA is a helper function to sign payloads with the ECDSA SignatureType
+func signECDSA(privKeyBytes, payload []byte, signatureType types.SignatureType) ([]byte, error) {
+	var sig []byte
+	var err error
+
+	switch signatureType {
+	case types.EcdsaRecovery:
+		sig, err = secp256k1.Sign(payload, privKeyBytes)
+		if err != nil {
+			return nil, fmt.Errorf("sign: unable to sign. %w", err)
+		}
+	case types.Ecdsa:
+		sig, err = secp256k1.Sign(payload, privKeyBytes)
+		if err != nil {
+			return nil, fmt.Errorf("sign: unable to sign. %w", err)
+		}
+		sig = sig[:64]
+	}
+	return sig, nil
+}
+
+// signEd25519 is a helper function to sign payloads with the Ed25519 SignatureType
+func signEd25519(privKeyBytes, payload []byte) []byte {
+	privKey := ed25519.NewKeyFromSeed(privKeyBytes)
```

Comment:
[From docs](https://golang.org/pkg/crypto/ed25519/#NewKeyFromSeed), this will panic if the private key is the wrong size, do we want to run the `KeyPair.isValid()` at some point before calling this to ensure we have a properly formatted key before using it here?